webapp/src/components/Metrics/PerformanceAnalysis.test.tsx

```diff
@@ -0,0 +1,138 @@
+import { fireEvent, screen, waitFor } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import PerformanceAnalysis from "./PerformanceAnalysis";
+import { AvailableDatasetSplits } from "types/api";
+import { setupServer } from "msw/node";
+import {
+  getCustomMetricInfoAPIResponse,
+  getMetricsPerFilterAPIResponse,
+  getMetricsPerFilterAPIWithFailureResponse,
+} from "mocks/api/mockMetricsAPI";
+
+const setDatasetSplitName = jest.fn();
+const renderPerformanceAnalysis = (
+  availableDatasetSplits: AvailableDatasetSplits
+) =>
+  renderWithRouterAndRedux(
+    <PerformanceAnalysis
+      jobId="local"
+      pipeline={{ pipelineIndex: 0 }}
+      availableDatasetSplits={availableDatasetSplits}
+      datasetSplitName="eval"
+      setDatasetSplitName={setDatasetSplitName}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+describe("PerformanceAnalysis", () => {
+  const handlers = [
+    getCustomMetricInfoAPIResponse,
+    getMetricsPerFilterAPIResponse,
+  ];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have two toggle buttons for the train and test", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    // verify if the toggles displayed with correct names
+    expect(
+      screen.getByRole("button", { name: "Evaluation Set", pressed: true })
+    ).toHaveValue("eval");
+    expect(
+      screen.getByRole("button", { name: "Training Set", pressed: false })
+    ).toHaveValue("train");
+  });
+
+  it("should display the expected columns", async () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    await waitFor(() => {
+      const expectedColumnHeaders = [
+        "Label",
+        "Total",
+        "Correct & Predicted",
+        "Correct & Rejected",
+        "Incorrect & Rejected",
+        "Incorrect & Predicted",
+        "Accuracy",
+        "Precision",
+        "Recall",
+        "F1",
+        "ECE",
+      ];
+      const actualColumnHeaders = screen.getAllByRole("columnheader");
+      expectedColumnHeaders.forEach((name, index) => {
+        name.includes("Correct")
+          ? expect(screen.getAllByTestId("CheckIcon")).toBeTruthy()
+          : name.includes("Incorrect")
+          ? expect(screen.getAllByTestId("XIcon")).toBeTruthy()
```

Comment:
You can check for the icons with `getByLabelText`, like you did in `OutcomeIcon.test.tsx`. Somethin like this:
```suggestion
        name.includes("&")
          ? screen.getByLabelText(name)
```

Reply:
We can even make it a little bit more robust by searching for the icon within the `actualColumnHeaders[index]`, with `within(actualColumnHeaders[index])` instead of in the whole `screen`.
```suggestion
        name.includes("&")
          ? within(actualColumnHeaders[index]).getByLabelText(name)
```
with
```diff
-import { fireEvent, screen, waitFor } from "@testing-library/react";
+import { fireEvent, screen, waitFor, within } from "@testing-library/react";
```

Reply:
Within makes it easier here with getbylabeltext :) Just adding getbylabeltext made the tests fail.

Reply:
Both suggestions work. I tried them before suggesting them. But yeah, `within` is more robust. Happy you like it. 👍